From 3513da4326fc58ab2d7fc6350bc883279ea65ee9 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 4 Jul 2024 10:15:26 +0200
Subject: [PATCH 1001/1695] dts: stm32: add firewall alternate conf property
 for omm on stm32mp2 boards

Add "access-controllers-conf-default" property that will supersede
"access-controllers" that should not be used to set a dynamic
configuration.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1239e4bc6b345b23f52f8831a243e43585cd6a3a dts: stm32: add firewall alternate conf property for omm on stm32mp2 boards]
Change-Id: I46a12cd836c6aa403fd281ba9cd82099fc013f8d
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391365
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi  | 6 ++++++
 core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi b/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi
index 0b73b5c2c..c685d34fa 100644
--- a/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi
+++ b/core/arch/arm/dts/stm32mp257f-dk-ca35tdcid-rif.dtsi
@@ -900,6 +900,12 @@
 		<&rifsc RIFPROT(STM32MP25_RIFSC_OCTOSPIM_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
 		<&rifsc RIFPROT(STM32MP25_RIFSC_OTFDEC1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
 		<&rifsc RIFPROT(STM32MP25_RIFSC_OTFDEC2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>;
+	access-controllers-conf-default =
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OCTOSPI1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OCTOSPI2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OCTOSPIM_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OTFDEC1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OTFDEC2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>;
 
 	st,omm-mux = <0x0>;
 
diff --git a/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi b/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi
index 92a0b61f9..b14e5381f 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi
+++ b/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi
@@ -900,6 +900,12 @@
 		<&rifsc RIFPROT(STM32MP25_RIFSC_OCTOSPIM_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
 		<&rifsc RIFPROT(STM32MP25_RIFSC_OTFDEC1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
 		<&rifsc RIFPROT(STM32MP25_RIFSC_OTFDEC2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>;
+	access-controllers-conf-default =
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OCTOSPI1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OCTOSPI2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OCTOSPIM_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OTFDEC1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>,
+		<&rifsc RIFPROT(STM32MP25_RIFSC_OTFDEC2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)>;
 
 	st,omm-mux = <0x0>;
 
-- 
2.47.2

